### ISSUE TITLE
Add missing required asterisk to title field’s placeholder

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#9028]
In our design system, the page title has a "required' asterisk next to its placeholder:

![Image](https://user-images.githubusercontent.com/877585/185309249-fe19740f-44c4-441f-b6e7-7f0941786c62.png)

This feels tricky to do in HTML/CSS, but it would be nice to find a way nonetheless. If possible without having to change the markup specifically for this field, so we keep field-specific customisations contained.

Currently, this title field’s custom styles are defined in: https://github.com/wagtail/wagtail/blob/9aeb2e3e49bb98201029d99fa613b0da9c6f1865/client/scss/components/forms/_title.scss#L34

To make matters worse, we currently support three widget types as title (CharField single-line, TextField multiline, RichTextField Draftail). It would be nice if we at least had the asterisk for the simplest single-line case.

**Comment by thibaudcolas** [comment]
Discussing this with Ben – we’re ok with the asterisk not being the correct color for the time being. So this will just include adding a ` *` at the end of the page title text here: https://github.com/wagtail/wagtail/blob/6814d0beea34ca3e868811210ff2210934e72ec3/wagtail/admin/panels.py#L1204

There might be tests to update as well but I’m not sure. This should make for a nice first contribution to Wagtail!

**Comment by maryojo** [comment]
Hi @thibaudcolas, I'm interested in working on this, can you assign it to me?

**Comment by damli40** [comment]
I'm interested as an outreachy applicant

**Comment by thibaudcolas** [comment]
Hi @maryojo @damli40, no need to ask! You can just go for it, first come first serve.

**Comment by maryojo** [comment]
Alright, I will work on it @thibaudcolas, thanks!

**Comment by Seremba** [comment]
Hi @thibaudcolas, I have worked on this issue. I am kindly requesting your review and suggestion on how to improve. 
The screenshot below shows some changes. Thanks a lot.
![Screenshot from 2022-10-13 18-34-48](https://user-images.githubusercontent.com/45463621/195642044-b35093b5-cc4c-48e3-848b-97353164bb6d.png)

**Comment by Dev-Yusuf** [comment]
Adding the asterisk is great. But how about the borderline? It's also important we make the border visible as others fields.

**Comment by gasman** [comment]
@Dev-Yusuf There's a separate issue about that - #9318

**Comment by Dev-Yusuf** [comment]
I will try doing something about that...

**Comment by thibaudcolas** [comment]
@maryojo it looks like @Seremba mistakenly worked on this as well and made a pull request addressing the problem. This isn’t ideal but I decided to accept the pull request nonetheless – mistakes happen, and there are plenty other good issues to look at.

I’ve made one and assigned it already to you @maryojo – it should be similar in scope to this one: https://github.com/wagtail/wagtail/issues/9363.

**Comment by maryojo** [comment]
Thanks @thibaudcolas! I was confused at first when I saw the PR

**Comment by Dev-Yusuf** [comment]
Please, I will be glad you assign me to any issues.
